common-lib/src/main/scala/com/gu/mediaservice/lib/elasticsearch6/ElasticSearchClient.scala

```diff
@@ -0,0 +1,120 @@
+package com.gu.mediaservice.lib.elasticsearch6
+
+import com.sksamuel.elastic4s.HealthStatus
+import com.sksamuel.elastic4s.http.ElasticDsl._
+import com.sksamuel.elastic4s.http.index.CreateIndexResponse
+import com.sksamuel.elastic4s.http.index.admin.IndexExistsResponse
+import com.sksamuel.elastic4s.http.{ElasticClient, ElasticProperties, Response}
+import play.api.Logger
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+
+trait ElasticSearchClient {
+
+  private val tenSeconds = Duration(10, SECONDS)
+  private val thirtySeconds = Duration(30, SECONDS)
+
+  def host: String
+
+  def port: Int
+
+  def cluster: String
+
+  def imagesAlias: String
+
+  protected val imagesIndexPrefix = "images"
+  protected val imageType = "image"
+
+  val initialImagesIndex = "images"
+
+  def shards: Int
+  def replicas: Int
+
+  lazy val client = {
+    Logger.info("Connecting to Elastic 6: " + host + " / " + port)
+    ElasticClient(ElasticProperties("http://" + host + ":" + port)) // TODO don't like this string config
+  }
+
+  def ensureAliasAssigned() {
+    Logger.info(s"Checking alias $imagesAlias is assigned to index…")
+    if (getCurrentAlias.isEmpty) {
+      ensureIndexExists(initialImagesIndex)
+      assignAliasTo(initialImagesIndex)
+      waitUntilHealthy()
+    }
+  }
+
+  def waitUntilHealthy(): Unit = {
+    Logger.info("waiting for cluster health to be green")
+    val clusterHealthResponse = Await.result(client.execute(clusterHealth().waitForStatus(HealthStatus.Green).timeout("25s")), thirtySeconds)
+    Logger.info("await cluster health response: " + clusterHealthResponse)
+    if (clusterHealthResponse.isError) {
+      throw new RuntimeException("cluster health could not be confirmed as green")  // TODO Exception isn't great but our callers aren't looking at our return value
+    }
+  }
+
+  def ensureIndexExists(index: String): Unit = {
+    Logger.info("Checking index exists…")
+
+    val eventualIndexExistsResponse: Future[Response[IndexExistsResponse]] = client.execute {
+      indexExists(index)
+    }
+
+    val indexExistsResponse = Await.result(eventualIndexExistsResponse, tenSeconds)
```

Comment:
I'd make this method return a `Future` and push the `Await` all the way up out of the client into the top-level calling code, which here I think is `MediaApiComponents` and `ThrallComponents`

Reply:
Agree that returning the Future to the calling code is sensible choose to try to match the existing Elastic1 behaviour which appears to be a no return and blocking.
The desire was to limit changes to the calling application but I'm ok to go either way on this.